### PR TITLE
Ana/fix dashboard substrate transactions

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

My PR yesterday (#4520) was missing some important things, which I hope to have solved here.

Working for Substrate transactions too (here there might be some technical aspects to discuss, but right now I think this is the way to do it)

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5f108ace63185b06d2757194--lunieio.netlify.app

File: api/lib/statistics.js [hunk 1]
```diff
@@ -22,16 +22,27 @@ const storeTransactions = (
...
-      hash: transaction.hash,
+      action: transaction.type || transaction.messageType,
+      hash: transaction.hash, // we are not getting this in Substrate networks for reasons explained below
       added: Date.now(),
       fingerprint
     }
-    if (!transaction.details.amounts && !transaction.details.amount) {
-      store(baseRow)
-      return
+    let amounts = []
```
Comment (faboweb) [inline api/lib/statistics.js:30]: how about:

```
const message = transaction.details || transaction.message
```

File: api/lib/statistics.js [hunk 2]
```diff
@@ -27,22 +27,14 @@ const storeTransactions = (
...
-        store(baseRow)
-        return
-      }
-      amounts = transaction.details.amounts || [transaction.details.amount]
+    const message = transaction.details || transaction.message
+    if (!message.amounts && !message.amount) {
+      store(baseRow)
+      return
     }
+    const amounts = message.amounts || [message.amount]
```
Comment (Bitcoinera) [inline api/lib/statistics.js:37]: Ah, looks better, true

**Comment by Bitcoinera** [comment]
Let's merge! :woman_cartwheeling:

**Pull request merged**